### ISSUE TITLE
Feature Request : add create tiddler function if 0 result searching

### ISSUE THREAD

**Issue by silvyn** [Jermolene/TiddlyWiki5#1801]
​Hello,

I don't know if this had ever discuss, but today I would like to have a link or button under search result in sidebar, which can propose me to create tiddler if the result is empty.

(before creating new tiddler, I check if I already have an entry whith the same content).

I think that will be in core.
What do you think ?

**Comment by sukima** [comment]
I love the idea, I've often thought about the same thing.

**Comment by sukima** [comment]
I think the line that drives the title is at [`core/modules/widgets/navigator.js:429`](https://github.com/Jermolene/TiddlyWiki5/blob/master/core/modules/widgets/navigator.js#L429)

**Comment by Jermolene** [comment]
I don't think this would need any JS mods; just a new tiddler button that uses the content of `$:/temp/search` as the tiddler title, and perhaps only reveals itself if the field is not empty. It could be added to the page controls toolbar.

**Comment by pmario** [comment]
hmm. If the search result field _is_ empty. right?

**Comment by pmario** [comment]
I did experiment with the `<$keyboard>` widget, eg: `ctrl+shift+enter` could trigger an action-widget event. .. but imo the `<$keyboard>` widget can't trigger a button event. ... At least I couldn't get it to work properly :/

**Comment by Jermolene** [comment]
We should definitely extend the keyboard widget to support action widgets. I think it would open the door to a few interesting shortcuts.

**Comment by Jermolene** [comment]
Just to be clear, we're now talking about two different solutions:

* A button, built on the existing core
* A keyboard shortcut, requiring an update to the keyboard widget

**Comment by sukima** [comment]
Should there be another ticket cut for the second line item?

**Comment by Jermolene** [comment]
Yes, please :)

**Comment by danielo515** [comment]
In my opinion this feature already existed , and we "bugfixed" it. Do you remember when the search widget returned a link to the searched term? If you clicked on it, you were asked for creating that tiddler.

**Comment by sukima** [comment]
@danielo515 can you elaborate? I don't see any thing close to that at tiddlywiki.com's search results. Clicking the search icon opens the "Advanced Search" tiddler and clicking the plus button opens a "New Tiddler".

**Comment by danielo515** [comment]
Hello @sukima

As I said, it is not in place anymore. It was on previous versions of TW. Try making a search on 5.1.5 for example. You will get a link to a non existing tiddler.

**Comment by pmario** [comment]
@silvyn ... I did create a little plugin: https://wikilabs.github.io/editions/new-tiddler-from-search/

Please test and close the issue if you are OK.

If not, please respond too!

**Comment by silvyn** [comment]
Hi @pmario and thanks for this proposal.
It's interesting, and I can do work with that.
Maybe the tag fields is not necessary here (or if we could use auto-completion it will be better, but not so simple, so I think I'll disable it to be sure of name tag (or just ignore it).

**Comment by pmario** [comment]
If the plugin is OK for you, you should close the issue.
